src/module-elasticsuite-core/Index/Mapping/Field.php

```diff
@@ -225,6 +225,14 @@ public function getDefaultSearchAnalyzer()
         return $this->config['default_search_analyzer'];
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    public function getConfig()
```

Comment:
Think it is a mistake to create a new getter for the config.
I would prefer a method mergeConfig(array $config) that would return a new instance of the field with updated config.

It is also possible to add a test on the mergeConfig method in FieldTest.